algorithms/unit_tests/TestStdAlgorithmsCompileOnly.cpp

```diff
@@ -110,6 +110,21 @@ struct TestStruct {
   Kokkos::DefaultExecutionSpace exe_space;
   std::string const label = "trivial";
 
+#ifndef KOKKOS_ENABLE_CXX17
+  static_assert(
+      std::random_access_iterator<
+          Kokkos::Experimental::Impl::RandomAccessIterator<Kokkos::View<T *>>>);
+  static_assert(std::random_access_iterator<
+                Kokkos::Experimental::Impl::RandomAccessIterator<
+                    Kokkos::View<const T *>>>);
+  static_assert(std::random_access_iterator<
+                Kokkos::Experimental::Impl::RandomAccessIterator<
+                    const Kokkos::View<T *>>>);
+  static_assert(std::random_access_iterator<
+                Kokkos::Experimental::Impl::RandomAccessIterator<
+                    const Kokkos::View<const T *>>>);
+#endif
```

Comment:
Book keeping: Create you own constexpr templated function that returns true, call it something like `test_kokkos_iterator_satify_std_random_access_iterator_concept()`, and static assert at namespace scope

Reply:
Here you go: c82e30e401dc34ccbd786bb51913d00ed921ae34